app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -0,0 +1,515 @@
+      dc_ids = clusters.collect { |c| c[:datacenter_id] }.uniq
+      storage_ids = []
+      dcs.each do |dc|
+        if dc_ids.include? dc.id
+          storage_ids << collector.collect_dc_domains(dc).to_miq_a.collect { |s| s[:id] }
+        end
+      end
+
+      persister_host = persister.hosts.find_or_build(host.id).assign_attributes(
+        :type             => 'ManageIQ::Providers::Redhat::InfraManager::Host',
+        :ems_ref          => ems_ref,
+        :ems_ref_obj      => ems_ref,
+        :name             => host.name || hostname,
+        :hostname         => hostname,
+        :ipaddress        => ipaddress,
+        :uid_ems          => host_id,
+        :vmm_vendor       => 'redhat',
+        :vmm_product      => host.type,
+        :vmm_version      => extract_host_version(host_os_version),
+        :vmm_buildnumber  => (host_os_version.build if host_os_version),
+        :connection_state => connection_state,
+        :power_state      => power_state,
+        :ems_cluster      => clusters,
+        :ipmi_address     => ipmi_address,
+        :storages         => storage_ids
+      )
+
+      host_operating_systems(persister_host, host, hostname)
+      networks = collector.collect_networks
+      switchs(persister_host, host, nics, networks)
+      host_hardware(persister_host, host, networks, nics)
+    end
+  end
+
+  def host_hardware(persister_host, host, networks, nics)
+    hdw = host.cpu
+
+    memory_total_attr = host.statistics.to_miq_a.detect { |stat| stat.name == 'memory.total' }
+    memory_total = memory_total_attr && memory_total_attr.dig(:values, :first, :datum)
+    hw_info = host.hardware_information
+
+    cpu_cores = hdw.dig(:topology, :cores) || 1
+    cpu_sockets = hdw.dig(:topology, :sockets) || 1
+
+    persister_hardware = persister.hosts.find_or_build(persister_host).assign_attributes(
+      :cpu_speed            => hdw.speed,
+      :cpu_type             => hdw.name,
+      :memory_mb            => memory_total.nil? ? 0 : memory_total.to_i / 1.megabyte,
+      :cpu_cores_per_socket => cpu_cores,
+      :cpu_sockets          => cpu_sockets,
+      :cpu_total_cores      => cpu_sockets * cpu_cores,
+      :manufacturer         => hw_info.manufacturer,
+      :model                => hw_info.product_name
+    )
+
+    host_networks(persister_hardware, nics)
+    host_guest_devices(persister_hardware, nics, networks)
+  end
+
+  def host_networks(persister_hardware, nics)
+    nics.to_miq_a.each do |vnic|
+      # Get the ip section
+      ip = vnic.ip.presence || {}
+
+      persister.host_nics.find_or_build(persister_hardware).assign_attributes(
+        :description => vnic.name,
+        :ipaddress   => ip.address,
+        :subnet_mask => ip.netmask,
+      )
+    end
+  end
+
+  def host_guest_devices(persister_hardware, nics, networks)
+    nics.to_miq_a.each do |nic|
+      network = network_from_nic(nic, host, networks)
+
+      if network
```

Comment:
switch_uid = network.try(:id) || network_name

should result in the same thing